src/UserGroup.elm

```diff
@@ -0,0 +1,130 @@
+module UserGroup exposing (UserGroup, Settings, DataRetentionPolicy, ContactDetails, Address, Tags, decoder)
+
+import Dict exposing (Dict)
+import Json.Decode exposing (..)
+
+
+type alias UserGroup =
+    { id : String
+    , parentId : String
+    , name : String
+    , children : List Child
+    , settings : Settings
+    , contactDetails : ContactDetails
+    , tags : Tags
+    }
+
+
+type alias Child =
+    { id : String
+    , name : String
+    }
+
+
+type alias Settings =
+    { inheritedFrom : Maybe String
+    , dataRetentionPolicy : DataRetentionPolicy
+    }
+
+
+type alias DataRetentionPolicy =
+    { idleDocTimeoutPreparation : Maybe Int
+    , idleDocTimeoutClosed : Maybe Int
+    , idleDocTimeoutCanceled : Maybe Int
+    , idleDocTimeoutTimedout : Maybe Int
+    , idleDocTimeoutRejected : Maybe Int
+    , idleDocTimeoutError : Maybe Int
+    , immediateTrash : Bool
+    }
+
+
+type alias ContactDetails =
+    { inheritedFrom : Maybe String
+    , address : Address
+    }
+
+
+type alias Address =
+    { preferredContactMethod : String
+    , email : Maybe String
+    , phone : Maybe String
+    , companyName : Maybe String
+    , address : Maybe String
+    , zip : Maybe String
+    , city : Maybe String
+    , country : Maybe String
+    }
+
+
+type alias Tags =
+    Dict String (Maybe String)
+
+
+decoder : Decoder UserGroup
+decoder =
+    map7 UserGroup
+        (field "id" string)
+        (field "parent_id" string)
+        (field "name" string)
+        (field "children" (list child))
+        (field "settings" settings)
+        (field "contact_details" contactDetails)
+        (field "tags" tags)
+
```

Comment:
Q: Why do you prefer `mapX` over applicative `andMap` from `Json.Decode.Extra` or things from `elm-json-decode-pipeline`? What are the benefits of this approach to you?

Reply:
The `mapX` functions work fine for the job. That's what they are for after all. Why add an unnecessary dependency when you don't need it? Unless you have more than 8 fields in your record you don't need to use the `andMap` style (and if you do have more than 8 you should probably consider simplifying/refactoring your record type anyway).

I get it that the applicative style can be easier to read and write for some. Especially if you are new to the language and don't have the intuition of what is going on with the decoders (i.e., that you are basically doing a function application on the `UserGroup` constructor function).